### PR TITLE
fix supporting Dict/OrderedDict.items/keys/values

### LINKED ISSUE #5341: DictPointer.items() can't be iterated
## Description
I got error when trying to iterate on `DictPointer.items()`, (while I can iterate on `ListPointer`).
## How to Reproduce
```python
# on DO side
duet.requests.add_handler(action="accept")

# on DS side
d = sy.lib.python.Dict({"1":1,"2":2})
dptr = d.send(duet)
itemsptr = dptr.items() # you get a ListPointer
itemsptr.__len__() # got error
for e in itemsptr: # got error too
    print(e)
```

## System Information
 - OS Version: [e.g. 22]
 - Language Version: [e.g. Python 3.7, Node 10.18.1]
 - Package Manager Version: [e.g. Conda 4.6.1, NPM 6.14.1]
 - Browser (if applicable): [e.g. Google Chrome]
 - Browser Version (if applicable): [e.g. 81.0.4044.138]

### PR DESCRIPTION
**Pull request by xutongye**
## Description
This may close #5341 .

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- dict_serde_test.py::test_iterator_methods
- ordered_dict_serde_test.py::test_iterator_methods

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: src/syft/lib/python/collections/ordered_dict.py [hunk 1]
```diff
@@ -102,13 +110,11 @@ def dict_get(self, other: Any) -> Any:
             # we can have torch.Tensor and other types
             return res
 
-    def items(self) -> SyPrimitiveRet:
-        res = list(super().items())
-        return PrimitiveFactory.generate_primitive(value=res)
+    def items(self, max_len: Optional[int] = None) -> Iterator:  # type: ignore
+        return Iterator(ItemsView(self), max_len=max_len)
```
Comment (tudorcebere) [inline src/syft/lib/python/collections/ordered_dict.py:114]: I think I like this more than to try to actually support ItemsView/KeysView

File: src/syft/lib/python/collections/ordered_dict.py [hunk 2]
```diff
@@ -102,13 +110,11 @@ def dict_get(self, other: Any) -> Any:
...
 
-    def keys(self) -> SyPrimitiveRet:
-        res = list(super().keys())
-        return PrimitiveFactory.generate_primitive(value=res)
+    def keys(self, max_len: Optional[int] = None) -> Iterator:  # type: ignore
+        return Iterator(KeysView(self), max_len=max_len)
```
Comment (tudorcebere) [inline src/syft/lib/python/collections/ordered_dict.py:117]: Same as above.

File: src/syft/lib/python/collections/ordered_dict.py [hunk 3]
```diff
@@ -130,9 +136,15 @@ def update(self, *args, **kwds: Any) -> SyPrimitiveRet:  # type: ignore
...
-    def values(self) -> SyPrimitiveRet:
-        res = list(super().values())
-        return PrimitiveFactory.generate_primitive(value=res)
+    def values(self, *args: Any, max_len: Optional[int] = None) -> Iterator:  # type: ignore
+        # this is what the super type does and there is a test in dict_test.py
+        # test_values which checks for this so we could disable the test or
+        # keep this workaround
+        if len(args) > 0:
+            traceback_and_raise(
+                TypeError("values() takes 1 positional argument but 2 were given")
```
Comment (tudorcebere) [inline src/syft/lib/python/collections/ordered_dict.py:145]: Could we replace 2 with `len(args)`?

**Review by tudorcebere — APPROVED** [review]

File: src/syft/lib/python/collections/ordered_dict.py (hunk 3 above)
Comment (xutongye) [reply src/syft/lib/python/collections/ordered_dict.py:145]: sure. I just took this code from dict.py

**Pull request merged**